### PR TITLE
report errors via JSON

### PR DESCRIPTION
**Pull request by eyakubovich**
This reflects the latest SPEC draft.

### CODE REVIEW

File: pkg/skel/skel.go [hunk 1]
```diff
@@ -87,12 +89,22 @@ func PluginMain(cmdAdd, cmdDel func(_ *CmdArgs) error) {
...
+
+func die(err string) {
+	plugin.PrintError(&plugin.Error{
+		Code: 100,
+		Msg:  err,
+	})
+	os.Exit(1)
+}
+
+func dief(f string, args ...interface{}) {
```
Comment (jonboulle) [inline pkg/skel/skel.go]: die and dief can just be one function with this signature

File: pkg/plugin/ipam.go [hunk 2]
```diff
@@ -127,7 +127,15 @@ func ConfigureIface(ifName string, res *Result) error {
 
 // PrintResult writes out prettified Result to stdout
 func PrintResult(res *Result) error {
-	data, err := json.MarshalIndent(res, "", "    ")
+	return prettyPrint(res)
+}
+
+func PrintError(err *Error) error {
```
Comment (jonboulle) [inline pkg/plugin/ipam.go]: uhh, stdout?

File: pkg/skel/skel.go [hunk 3]
```diff
@@ -61,13 +64,12 @@ func PluginMain(cmdAdd, cmdDel func(_ *CmdArgs) error) {
...
 	if argsMissing {
-		os.Exit(1)
+		die("required env variables missing")
 	}
 
 	stdinData, err := ioutil.ReadAll(os.Stdin)
 	if err != nil {
-		log.Printf("Error reading from stdin: %v", err)
-		os.Exit(1)
+		dief("Error reading from stdin: %v", err)
```
Comment (jonboulle) [inline pkg/skel/skel.go]: inconsistent capitalisation at start of die messages

File: pkg/plugin/ipam.go (hunk 2 above)
Comment (jonboulle) [inline pkg/plugin/ipam.go]: docstring like PrintResult

**Comment by jonboulle** [comment]
few nits but LGTM

File: pkg/skel/skel.go (hunk 1 above)
Comment (eyakubovich) [inline pkg/skel/skel.go]: you're right. there's an edge case but it's private so it'll be fine.

**Pull request merged**